src/foam/mlang/mlang.js

```diff
@@ -2867,28 +2868,37 @@ foam.CLASS({
         // of parameter is an interface rather than a class.
         return a;
       },
+      javaJSONParser: 'foam.lib.json.ExprParser.instance()',
       name: 'tail'
-    },
-    {
-      name: 'compare',
-      swiftSupport: false,
-      transient: true,
-      documentation: 'Is a property so that it can be bound to "this" so that it works with Array.sort().',
-      factory: function() { return this.compare_.bind(this); }
     }
   ],
 
   methods: [
-    function compare_(o1, o2) {
-      // an equals of arg1.compare is falsy, which will then hit arg2
-      return this.head.compare(o1, o2) || this.tail.compare(o1, o2);
+    {
+      name: 'compare',
+      code: function(o1, o2) {
+        // an equals of arg1.compare is falsy, which will then hit arg2
+        return this.head.compare(o1, o2) || this.tail.compare(o1, o2);
+      },
+      javaCode: `return getHead().compare(o1, o2) == 0 ? getTail().compare(o1, o2) : getHead().compare(o1, o2);`
```

Comment:
This is inefficient because it might have to call getHead().compare(o1, o2) twice. Better:
int ret = getHead().compare(o1, o2);
return ret == 0 ? getTail().compare(o1, o2) : ret;

Reply:
@kgrgreer fixed